contrib/lib/src/databases.rs

```diff
@@ -848,7 +848,7 @@ impl<K, C: Poolable> Drop for Connection<K, C> {
                 // Explicitly dropping the permit here so that it's only
                 // released after the connection is.
                 drop(permit);
-            })
+            }).await
```

Comment:
This...is interesting. I don't believe this does anything. @jebrosen: do you have any additional insight?

Reply:
This change was made to fix the error [async_yields_async](https://rust-lang.github.io/rust-clippy/master/index.html#async_yields_async). It think this lint is to avoid code like this one :
```
let x = async {
    other_async_fn()
  };
x.await
  .await
```

Reply:
The goal of this awkwardness is to run `async` and blocking code in the appropriate contexts; it doesn't actually need to be waited on. If `;` silences the lint that might even be better!

Adding this `.await` will likely keep one task alive a little longer than necessary on every `Drop`.

Reply:
Good call @jebrosen! The semicolon was missing, not the await.

Clippy's lint and their explanations are not always good, but they have the merit to make us think and to make us learn new stuff. Thanks for the explanation. :)